.github/workflows/codeql-tests.yaml

```diff
@@ -3,6 +3,9 @@ name: 'CodeQL Tests'
 
 on:
   push:
+    branches-ignore:
```

Comment:
Does that mean any other branch runs the tests? I wonder if we should only run the tests on specific branches (e.g. `dev`, `main`)?

Reply:
> Does that mean any other branch runs the tests? I wonder if we should only run the tests on specific branches (e.g. `dev`, `main`)?

Currently these checks run on all branches. We want these to run on arbitrary branches so PR code will be checked, but it's unnecessary to run them for e2e runs.

Reply:
Good feedback, I updated the PR to include these suggestions.